zerver/actions/message_send.py

```diff
@@ -1710,13 +1712,21 @@ def check_message(
 
     if (
         stream is not None
-        and message_send_dict.rendering_result.has_wildcard_mention()
-        and not wildcard_mention_allowed(sender, stream, realm)
+        and message_send_dict.rendering_result.mentions_stream_wildcard
+        and not stream_wildcard_mention_allowed(sender, stream, realm)
     ):
         raise JsonableError(
-            _("You do not have permission to use wildcard mentions in this stream.")
+            _("You do not have permission to use stream wildcard mentions in this stream.")
         )
```

Comment:
We should probably turn this into a parallel `StreamWildcardMentionNotAllowedError` thing in `exceptions.py` just so that these work parallel to each other -- I can see mobile and other clients wanting to do special handling of this exception in some situations.

Reply:
Done